### PR TITLE
gazebo_ros_camera: configure projection matrix from sdf

### PR DESCRIPTION
**Pull request by ihasdapie**
This PR allows for the projection matrix in `gazebo_ros_camera` to be configured from SDF.

Signed-off-by: Brian Chen <brian.chen@openrobotics.org>

### CODE REVIEW

File: gazebo_plugins/src/gazebo_ros_camera.cpp
Comment (jacobperron) [inline gazebo_plugins/src/gazebo_ros_camera.cpp]: Is a single-quote character valid in SDF? I'm trying to imagine what setting this in practice looks like,

```xml
<cx'>100</cx'>
```

I would make the names clear that they're specifically for overriding the projection matrix, P, e.g. `P_cx`, `P_cy`, `P_fx`, `P_fy`, etc.

**Comment by jacobperron** [comment]
For posterity, the idea behind this PR is to allow users to configure the projection matrix, P, independently of the intrinsics matrix, K. I think this makes sense in the case of stereo cameras, where P may be different from K due to the transform between the two cameras.

File: gazebo_plugins/src/gazebo_ros_camera.cpp
Comment (ihasdapie) [reply gazebo_plugins/src/gazebo_ros_camera.cpp]: Yup that'd be a good idea. I've pushed changes to reflect that.

File: gazebo_plugins/src/gazebo_ros_camera.cpp
Comment (jacobperron) [inline gazebo_plugins/src/gazebo_ros_camera.cpp]: We're also interested in exposing the R matrix parameters above. I'm not sure what is the best way to set a matrix from SDF.

Comment (ihasdapie) [reply gazebo_plugins/src/gazebo_ros_camera.cpp]: It appears that in gazebo classic and gazebo matrices are set elementwise with a new sdf tag for each settable entry. 
Unless there's a better way we could just do this.

```suggestion
    camera_info_msg.r[0] = _sdf->Get<double>("r_0", 1.0).first;
    camera_info_msg.r[1] = _sdf->Get(double>("r_1", 0.0).first
    // ... and so forth
```

Comment (jacobperron) [reply gazebo_plugins/src/gazebo_ros_camera.cpp]: Maybe @adityapande-1995 or @scpeters can recommend an approach.

Comment (ihasdapie) [reply gazebo_plugins/src/gazebo_ros_camera.cpp]: I _think_ we could also do something like this:
`<rectification_matrix>1 0 0 0 1 0 0 0 1</rectification_matrix>`

And while we're at it make the other matricies use this syntax as well.

Comment (scpeters) [reply gazebo_plugins/src/gazebo_ros_camera.cpp]: > I _think_ we could also do something like this: `<rectification_matrix>1 0 0 0 1 0 0 0 1</rectification_matrix>`
> 
> And while we're at it make the other matricies use this syntax as well.

yes, I think this would be easier than specifying all the matrix parameters individually, as long as we document the order of parameters. I recommend this approach

Comment (scpeters) [reply gazebo_plugins/src/gazebo_ros_camera.cpp]: should we block this until we add the `rectification_matrix` or merge as is?

Comment (ihasdapie) [reply gazebo_plugins/src/gazebo_ros_camera.cpp]: I think it is reasonable to merge as-is and then we can add `rectification_matrix` and the other matricies in a new PR

**Comment by scpeters** [comment]
this looks good to me; we just need to add documentation of the SDFormat parameters to [gazebo_ros_camera.hpp](https://github.com/ros-simulation/gazebo_ros_pkgs/blob/ros2/gazebo_plugins/include/gazebo_plugins/gazebo_ros_camera.hpp) and example usage in [gazebo_ros_camera_demo.world](https://github.com/ros-simulation/gazebo_ros_pkgs/blob/ros2/gazebo_plugins/worlds/gazebo_ros_camera_demo.world)

**Comment by ihasdapie** [comment]
> this looks good to me; we just need to add documentation of the SDFormat parameters to [gazebo_ros_camera.hpp](https://github.com/ros-simulation/gazebo_ros_pkgs/blob/ros2/gazebo_plugins/include/gazebo_plugins/gazebo_ros_camera.hpp) and example usage in [gazebo_ros_camera_demo.world](https://github.com/ros-simulation/gazebo_ros_pkgs/blob/ros2/gazebo_plugins/worlds/gazebo_ros_camera_demo.world)

@scpeters Resolved in  https://github.com/ros-simulation/gazebo_ros_pkgs/pull/1409/commits/1470786adc393e771f9ae60c4907d81d51cf5599. 

The values I put in for the the example usage and in the demo world are the same as before i.e. the default calculated values using the existing tags so no behaviour changes.

File: gazebo_plugins/include/gazebo_plugins/gazebo_ros_camera.hpp
Comment (scpeters) [inline gazebo_plugins/include/gazebo_plugins/gazebo_ros_camera.hpp:69]: can you also add `Ty` and `P_fy` and a brief description of each parameter?

File: gazebo_plugins/src/gazebo_ros_camera.cpp
Comment (kjeppesen1) [reply gazebo_plugins/src/gazebo_ros_camera.cpp]: Here is the `rectification_matrix` PR: https://github.com/ros-simulation/gazebo_ros_pkgs/pull/1423

File: gazebo_plugins/include/gazebo_plugins/gazebo_ros_camera.hpp
Comment (scpeters) [reply gazebo_plugins/include/gazebo_plugins/gazebo_ros_camera.hpp:69]: 24cf9b721557b046227bbe015d35ecca1bf79b7b, thanks

**Review by scpeters — APPROVED** [review]

**Pull request merged**
